test/adapter/json_api/relationships_test.rb

```diff
@@ -158,10 +157,16 @@ def test_relationship_meta
           end
 
           def test_relationship_not_including_data
+            @author.define_singleton_method(:read_attribute_for_serialization) do |attr|
+              fail 'should not be called' if attr == :blog
```

Comment:
this test needs to change as well.  It shouldn't be raising a runtime error  and is hard to read and why do we even need the define_singleton_method?

😦